tests/backup/backup_restore_basic_test.go

```diff
@@ -895,3 +896,197 @@ var _ = Describe("{CustomResourceRestore}", func() {
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						err = CreateBackup(backupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace},
+							labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation: %s", backupName))
+					}(backupName)
+				}
+				wg.Wait()
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Remove the backup location where backups were taken", func() {
+			log.InfoD("Remove backup location where backups were taken")
+			err := DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID, false)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
+			//TODO: Eliminate time.Sleep
+			time.Sleep(time.Minute * 3)
+		})
+
+		Step("Add the backup location again which had backups", func() {
+			log.InfoD("Add the backup location with backups back")
+			for _, provider := range providers {
+				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				err := CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+			}
+		})
+		Step("Taking backup of applications to trigger BackupSync goroutine", func() {
+			log.InfoD("Taking backup of applications to trigger BackupSync goroutine")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			for _, namespace := range bkpNamespaces {
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				backupNamespaceMap[namespace] = backupName
+				err = CreateBackup(backupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace}, labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation %s", backupName))
+				backupNames = append(backupNames, backupName)
+			}
+		})
+
+		Step("Check if all backups are synced or not", func() {
+			log.InfoD("Check if backups created before are synced or not")
+
+			// Wait for backups to get synced
+			for i := 0; i < 1000; i++ {
+				fetchedBackupNames, _ := GetAllBackupsAdmin()
+				log.Infof(fmt.Sprintf("Fetched backups %d", len(fetchedBackupNames)))
+				if len(fetchedBackupNames) == len(backupNames) {
+					break
+				} else {
+					time.Sleep(30)
+				}
+			}
+			fetchedBackupNames, err := GetAllBackupsAdmin()
+			log.FailOnError(err, "Getting a list of all backups")
+			dash.VerifyFatal(len(fetchedBackupNames), len(backupNames), "Comparing the expected and actual number of backups")
```

Comment:
Ideally It would be better if all the other cases cleanup their backups, the problem here which can happen is that we would be able to validate if backupsync is able to pull in all the backups or not. And this thing might not happen as well because by default  `DeleteBackupLocation()` would run with true and remove the backups present in the bucket.